### DISCUSSION TITLE
react-native, axios, and aws application load balancer

### DISCUSSION THREAD

**Discussion by mblarsen** [axios/axios#6019] [Q&A]
I'm posting here since not sure where else to start and given the clues shown below it seems the most sensible place to start query.

We are sending a cookie using Axios to our API something like this:

```
axios.post(url, { ... some data ... }, { 
	headers: { Cookie: "my_cookie=foo" }
});
```

and in the backend we get a corrupted header that looks something like this. 

```
cookie: AWSLB=abc; AWSOTHER=def; AWSOTHER2=ghi,my_cookie=foo
```

This is just an example, we have other cases where get complete absurd cookie content like something relating to Shopify:

<details>

```
_S=69714497-bba2-4465-a668-a47fd809df79;
_shopify_s=6xxxxxxx-bba2-4465-a668-a47fd809df79;
_shopify_y=4xxxxxxx-fb6d-4a24-8a9a-49266745£188; _y=4241c653-fb6d-4a24-8a9a-4926
6745£188; _tracking_consent=87B%22region2283A822PH0082282C822reg82283A82282282C822con%2283A87B%22CMP%2283A87B%22a82283A822%2282C822582283A82282282C822p82283A82282282C822m&2283A8
22%2287087082C822v82283A8222.182272C8221im%2283A85B%22GDPR_BLOCK_ALL%22%5D7D,my_cookie=foo
```
</details>

What we have looked into:
1. the cookie value doesn't include a leading comma `,`
2. that our API doesn't mess up the cookie in decoding or otherwise. Used `ncat` to inspect incoming headers which indeed are corrupted before reaching the app.
3. tried with and without stickiness in the application load balancer, no difference
4. tried disabling and enabling removal of bad headers

Are there a way to inspect the headers (e.g. in intercepter) right before axios sends the request?

Also note we are running in react-native with some extensions like Didomi, Firebase, Amplitude that could maybe alter the request before sending to ensure GDPR because we do not add this ourselves. I'm not aware of this, just speculation. 

Other feedback is welcome. Thanks

ps: we are in communication with AWS regarding the load-balancer but so far no result.

**Comment by jasonsaayman** [comment]
In React Native, Axios does not ultimately write the raw socket headers itself; it delegates to React Native's networking layer. A request interceptor can show you the Axios config before handoff:

```js
axios.interceptors.request.use(config => {
  console.log('axios headers:', config.headers?.toJSON?.() ?? config.headers);
  return config;
});
```

But that only confirms what Axios is passing down. If the header is changed after that, it would be in the React Native/native networking layer, a library hooking requests, or infrastructure.

One important note: manually setting the `Cookie` header from app code is not reliable across browser-like/native environments. Cookie handling is often managed by the platform cookie jar, and multiple cookie sources may be merged. If possible, prefer using the platform/session cookie mechanism instead of sending `headers: { Cookie: ... }` manually.

For debugging, I would compare:

- The interceptor output before Axios hands off the request.
- A minimal React Native app with no Didomi/Firebase/Amplitude/network instrumentation.
- The same request from Node.js or curl directly to the ALB.
- What arrives at a raw listener before your application parses cookies.

If Axios logs only `my_cookie=foo` but the raw listener receives other cookies, the mutation is happening after Axios or at the platform/infrastructure layer.
